ui/src/app/workflows/components/workflow-dag/workflow-dag.tsx

```diff
@@ -390,7 +397,56 @@ export class WorkflowDag extends React.Component<WorkflowDagProps, WorkflowDagRe
         return {nodes, edges};
     }
 
-    private layoutGraph(nodes: string[], edges: Edge[]) {
+    private layoutGraphPretty(nodes: string[], edges: Edge[]) {
+        const graph = new dagre.graphlib.Graph();
+
+        graph.setGraph({
+            edgesep: 20 / this.scale,
+            nodesep: 50 / this.scale,
+            rankdir: this.state.horizontal ? 'LR' : 'TB',
+            ranksep: 50 / this.scale
+        });
+
+        graph.setDefaultEdgeLabel(() => ({}));
+
+        nodes.forEach(node => {
```

Comment:
would it be faster to combine these two foreachs into one?

Reply:
Not sure what you mean, one iterates over nodes and the other over edges. We need to look at both as they are independent?